tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp

```diff
@@ -75,6 +78,57 @@ tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
     return result;
 }  // ecall_GenerateNonce
 
+/* Verify unique id singature obtained from KME by using unique id key.
+ * unique id signature is digital signature computed on the hash value
+ * of concated string of unique id and WPE generated nonce.
+ *
+ * @param unique_id_key - Public key generated by KME to be
+ *              used for registration
+ * @param verification_key_signature - digital signature computed on
+ *              hash of concatenated string of unique id and nonce
+ */
+tcf_err_t ecall_VerifyUniqueIdSignature(const char* unique_id_key,
+    const char* verification_key_signature) {
+
+    tcf_err_t result = TCF_SUCCESS;
+    try {
+        tcf::error::ThrowIfNull(unique_id_key, "Unique ID is null");
+        tcf::error::ThrowIfNull(verification_key_signature,
+            "Verification key signature is null");
+
+        ByteArray verify_key_hex_bytes = HexEncodedStringToByteArray(
+            unique_id_key);
+        std::string b64_verify_key = ByteArrayToStr(verify_key_hex_bytes);
+        tcf::crypto::sig::PublicKey public_key(b64_verify_key);
+
+        EnclaveData* enclave_data = EnclaveData::getInstance();
+        // compute hash on b64 encoded verification key and nonce hex string
+        std::string concat_str = b64_verify_key + enclave_data->get_nonce();
+        ByteArray hash = tcf::crypto::ComputeMessageHash(
+            StrToByteArray(concat_str));
+        std::string b64_verify_key_sig = ByteArrayToStr(
+            HexEncodedStringToByteArray(verification_key_signature));
```

Comment:
`b64_verify_key_sig`  is unused. Please remove

Reply:
Removed.